### PR TITLE
Small Fix

### PR DESCRIPTION
**Pull request by nrummel**
Hello JSO Team,
 
I wanted to use your ARCqK algorithm in my work, but it was cycling when my cost function was misbehaving. I think the algorithm is supposed to exit when there are too many unsuccessful iterations in a row, but instead it got stuck in an infinite loop until time controls hit.
Also, I want to pass the nlp_at_x the callback so that I have access to the current x at each iteration. 
Finally, I think the counter for very successful iterations was in the wrong place. 

Thanks for making your algorithm available in Julia.
Nic

### CODE REVIEW

**Comment by tmigot** [comment]
Hi @nrummel ! Thanks for the PR, I think this is connect to this issue: https://github.com/JuliaSmoothOptimizers/AdaptiveRegularization.jl/issues/84

File: src/main.jl [hunk 1]
```diff
@@ -148,8 +148,8 @@ function SolverCore.solve!(
     [Int64, T, T, T, String, T, T, T],
   )
   verbose > 0 && @info log_row(Any[iter, ft, norm_∇f, 0.0, "First iteration", α])
-
-  callback(nlp, solver, stats)
```
Comment (tmigot) [inline src/main.jl:152]: You can actually access the nlp_at_x by solver.stp.nlp_at_x , so I prefer `callback(nlp, solver, stats)`

File: src/main.jl [hunk 2]
```diff
@@ -148,8 +148,8 @@ function SolverCore.solve!(
...
+  # Pass the nlp_at_x so that we have access to xᵢₜₑᵣ
+  callback(nlp_at_x, solver, stats)
```
Comment (tmigot) [inline src/main.jl:152]:
```suggestion
  callback(nlp, solver, stats)
```

File: src/main.jl [hunk 3]
```diff
@@ -237,7 +235,11 @@ function SolverCore.solve!(
     set_dual_residual!(stats, nlp_at_x.current_score)
     set_iter!(stats, nlp_stop.meta.nb_of_stop)
     set_time!(stats, nlp_at_x.current_time - nlp_stop.meta.start_time)
-    callback(nlp, solver, stats)
+    if !success & (unsuccinarow >= max_unsuccinarow)
+      # don't just cycle if we have to many unsucessful iterations 
+      stats.status = :user
+    end
+    callback(nlp_at_x, solver, stats)
```
Comment (tmigot) [inline src/main.jl:242]:
```suggestion
    callback(nlp, solver, stats)
```

File: src/main.jl [hunk 4]
```diff
@@ -205,25 +205,23 @@ function SolverCore.solve!(
...
         norm_∇f = norm(∇f)
-
-        verysucc += 1
         if r > increase_threshold # very sucessful
           α = increase(PData, α, TR)
           verbose > 0 &&
-            mod(iter, verbose) == 0 &&
+            mod(iter, verbose) == 0 && 
             @info log_row(Any[iter, ft, norm_∇f, λ, "V", α, norm(d), Δq])
+          verysucc += 1 # TODO: check if this makes sense
```
Comment (tmigot) [inline src/main.jl:213]:
```suggestion
          verysucc += 1
```
looks good, thanks

File: src/main.jl [hunk 5]
```diff
@@ -237,7 +235,11 @@ function SolverCore.solve!(
     set_dual_residual!(stats, nlp_at_x.current_score)
     set_iter!(stats, nlp_stop.meta.nb_of_stop)
     set_time!(stats, nlp_at_x.current_time - nlp_stop.meta.start_time)
-    callback(nlp, solver, stats)
+    if !success & (unsuccinarow >= max_unsuccinarow)
+      # don't just cycle if we have to many unsucessful iterations
```
Comment (tmigot) [inline src/main.jl:239]: I don' think the scenario, `success = true` and `unsuccinarow >= max_unsuccinarow` is possible.
```suggestion
    if unsuccinarow >= max_unsuccinarow
```

**Review by tmigot — CHANGES_REQUESTED** [review]
Thanks for the PR @nrummel ! There are still improvements that can be made to this algorithm, so thank you for testing and contributing.

I made some comments, but overall it looks good.

**Comment by nrummel** [comment]
I still am not seeing the remote/main having these changes. It would be nice if this was merged into main, so I don't have to dev this repo anymore. Thanks again. Nic

**Comment by tmigot** [comment]
Hi @nrummel ! As you can see here, I made some comments that you need to adress before the PR is being merged.

**Comment by nrummel** [comment]
Thanks for doing this. Apologies, I have been pushing this small task down my to do list for a while now.

**Pull request merged**